src/components/serviceProviderSearch/ServiceProviderSearch.js

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import ServicesCategoryService from "../../services/ServiceCategoryService";
+import mockCategory from "../../mock/data/service.providers.mock";
+
+export default class ServicesProviderSearch extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.serviceCategoryService = ServicesCategoryService.getInstance()
+        this.state = {
+            serviceCategory: {
+                title: '',
+                id: 1
+            }
+        }
+    }
+
+    componentDidMount() {
+        this.serviceCategoryService.findServiceCategoryById(this.props.match.params.id)
+            .then(serviceCategory => {
+                    this.setState({
+                        serviceCategory: serviceCategory
+                    });
+                    console.log(this.state.serviceCategory)
+                }
+            ).catch(err => {
+                //We'll use mock data
+                this.setState({
+                    serviceCategory: mockCategory[0]
+                });
+                console.log("Using Mocks!");
+            })
+    }
+
+    generateFilters() {
+        let h5s = [];
+        if (this.state.serviceCategory.questions != null)
+        for (var i = 0; i < this.state.serviceCategory.questions.length; i++) {
+            let filter  =
+                <div key={i + " question"}>
+                    <h5>{ this.state.serviceCategory.questions[i].question}</h5>
+                    { this.generateChoices(i) }
+                </div>;
+
+            h5s.push(filter);
+        }
+        return h5s;
+    }
+
+    generateProviders() {
+        let h5s = [];
+        if (this.state.serviceCategory.questions != null)
+            for (var i = 0; i < this.state.serviceCategory.serviceProviders.length; i++) {
+                var serviceProvider = this.state.serviceCategory.serviceProviders[i];

```

Comment:
This might be a very dumb question, but does this actually work with the data model that we have? I'm not too familiar with how the ServiceCategory data model was set up, so it might, but it seems like this code is very heavily reliant on the ServiceCategory to have all the data we need.

Reply:
Either way, you and I are going to need to work together on a follow-up PR to integrate the Home Screen SearchBar and this code, since the search needs to redirect here. We can worry about that after, though.